packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1189,6 +1458,47 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
     return this.newInstancesProtectedFromScaleIn === true;
   }
 
+  /**
+   * Allows specify security group connections for instances of this fleet.
+   */
+  public get connections(): ec2.Connections {
+    if (this._connections) {
+      return this._connections;
+    }
+
+    if (this.launchTemplate) {
+      return this.launchTemplate.connections;
+    }
+
+    throw new Error('AutoScalingGroup can only be used as IConnectable if it is not created from an imported Launch Template.');
+  }
+
+  /**
+   * UserData for the instances
+   */
```

Comment:
Can we improve this comment to explain what the `UserData` does and how it is used?

Reply:
I can borrow some words from the public doc. However more importantly I think it's better to document the fact that this getter can throw error. Same as the `role` getter.